targets/XPlatCppSdk/source/test/TestApp/TestContext.cpp

```diff
@@ -8,6 +8,29 @@ namespace PlayFabUnit
 {
     void TestContext::EndTest(TestFinishState state, std::string resultMsg)
     {
+        if (finishState != TestFinishState::PENDING)
```

Comment:
If the status is not Pending that means that we already finish the test at least once, and we are running it again  for a different reason.